Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -1407,7 +1407,64 @@ bool ObjectNode::BuildArgs( const Job * job, Args & fullArgs, Pass pass, bool us
                 continue;
             }
         }
-				
+
```

Comment:
This is a good fix, but it's a little unfortunate that:
a) we perform this logic redundantly for every obj file (potentially hundreds or thousands of times in a build).  Of particular note is that EnsurePathExists is a call into the kernel.
b) we do it in BuildArgs, which isn't really a function that should have side-effects

Ideally, we'd have this logic performed once by the ObjectList or Library that owns the ObjectNodes, so we'd only have to do it once.  Sadly, I don't think there's a good way to do that right now.  I'll need to give it some more thought.

I think I can integrate this as-is though, and look into options for reducing the overhead.